### PR TITLE
Install specific version of mailcatcher

### LINKED ISSUE #5410: [catalog-staging][catalog-qa] Mailcatcher doesn't catch mail on qa and staging
### Expected behavior
When I submit a request through the request form in catalog-qa or catalog-staging I expect mailcatcher to catch the mail.

### Actual behavior
Mailcatcher does not catch the mail. 

### Steps to replicate:
- Tunnel to the catalog-staging mailcatcher:
`ssh -L 1082:localhost:1080 pulsys@catalog-staging1`

- Go to https://catalog-staging.princeton.edu/catalog/99118269913506421
- Click on the 'Request' button. Login through CAS
- Select 'Physical Item Delivery'
- Click on 'Request this item'

- Go to the [Orangelight project in honeybadger](https://app.honeybadger.io/projects/54399/faults?q=-is%3Aresolved+-is%3Aignored&sort=last_seen_desc)
You will see the same error like the ones below, login your email . It means that mailacatcher didn't catch the email and staging tried to send an email to Relay and Relay denied as expected. 

- Go to localhost:1082 (it's the mailcatcher on staging). You can see that your request is not in the mailcatcher.

### Honeybadger link and code snippet, if applicable
Related honeybadger errors: 
https://app.honeybadger.io/projects/54399/faults/113222384
https://app.honeybadger.io/projects/54399/faults/111757571
https://app.honeybadger.io/projects/54399/faults/113222385

### PR DESCRIPTION
**Pull request by maxkadel**
- Reload daemon when configuration changes
- Helps keep install path consistent with mailcatcher version

Closes #5410

### CODE REVIEW

File: group_vars/pdc_describe/staging.yml [hunk 1]
```diff
@@ -24,7 +24,8 @@ pdc_describe_production_aws_dspace_bucket: 'prds-dataspace'
 install_mailcatcher: true
 mailcatcher_user: "pulsys"
 mailcatcher_group: "pulsys"
-mailcatcher_install_location: "/usr/local/lib/ruby/gems/3.1.0/gems/mailcatcher-0.8.2/bin/mailcatcher"
+mailcatcher_version: 0.8.2
```
Comment (hectorcorrea) [inline group_vars/pdc_describe/staging.yml:27]: 👍🏼

File: group_vars/oawaiver/staging.yml [hunk 2]
```diff
@@ -6,7 +6,8 @@ running_on_server: true
 install_mailcatcher: true
 mailcatcher_user: "pulsys"
 mailcatcher_group: "pulsys"
-mailcatcher_install_location: "/usr/local/lib/ruby/gems/3.1.0/gems/mailcatcher-0.10.0/bin/mailcatcher"
+mailcatcher_version: 0.10.0
```
Comment (hectorcorrea) [inline group_vars/oawaiver/staging.yml:9]: 👍🏼

File: group_vars/pdc_discovery/staging.yml [hunk 3]
```diff
@@ -18,7 +18,8 @@ pdc_discovery_honeybadger_key: '{{vault_pdc_discovery_honeybadger_key}}'
 install_mailcatcher: true
 mailcatcher_user: "pulsys"
 mailcatcher_group: "pulsys"
-mailcatcher_install_location: "/usr/local/lib/ruby/gems/3.3.0/gems/mailcatcher-0.10.0/bin/mailcatcher"
+mailcatcher_version: 0.10.0
```
Comment (hectorcorrea) [inline group_vars/pdc_discovery/staging.yml:21]: 👍🏼

File: group_vars/tigerdata/qa.yml [hunk 4]
```diff
@@ -18,7 +18,8 @@ tigerdata_honeybadger_key: '{{ vault_honeybadger_api_key }}'
 install_mailcatcher: true
 mailcatcher_user: "pulsys"
 mailcatcher_group: "pulsys"
-mailcatcher_install_location: "/usr/local/lib/ruby/gems/3.1.0/gems/mailcatcher-0.9.0/bin/mailcatcher"
+mailcatcher_version: 0.9.0
+mailcatcher_install_location: "/usr/local/lib/ruby/gems/3.1.0/gems/mailcatcher-{{ mailcatcher_version }}/bin/mailcatcher"
```
Comment (hectorcorrea) [inline group_vars/tigerdata/qa.yml:22]: 👍🏼

File: group_vars/tigerdata/staging.yml [hunk 5]
```diff
@@ -19,7 +19,8 @@ tigerdata_honeybadger_key: '{{ vault_honeybadger_api_key }}'
 install_mailcatcher: true
 mailcatcher_user: "pulsys"
 mailcatcher_group: "pulsys"
-mailcatcher_install_location: "/usr/local/lib/ruby/gems/3.1.0/gems/mailcatcher-0.9.0/bin/mailcatcher"
+mailcatcher_version: 0.9.0
```
Comment (hectorcorrea) [inline group_vars/tigerdata/staging.yml:22]: 👍🏼

**Review by hectorcorrea — APPROVED** [review]

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel 🦄

**Pull request merged**